Sources/NIO/EventLoop.swift

```diff
@@ -463,6 +467,8 @@ extension EventLoop {
     /// - returns: An `EventLoopFuture` containing the result of `task`'s execution.
     @inlinable
     public func submit<T>(_ task: @escaping () throws -> T) -> EventLoopFuture<T> {
+        precondition(self.isOpen, "cannot submit work to a closed EventLoop")
```

Comment:
`isOpen` here has some undocumented thread safety assumptions that I suspect it fails to meet. In fact, the implementation on `MultithreadedEventLoopGroup` is definitely thread _unsafe_ for this usage model. Remember that `submit` is valid to call from any thread, so it cannot safely read data that is unsynchronised.

In this case, I think the design cannot solve this problem generically: it's a requirement of the event loop implementation specifically to solve this problem. This is because the baseline function that needs to be correct is `eventLoop.execute`, which is entirely up to the individual event loop to implement. We therefore can only fix it for `MultithreadedEventLoopGroup`, and document that other event loops should also endeavour to meet this requirement.

Does that make sense?